notes-20200211.tex

```diff
@@ -0,0 +1,182 @@
+\documentclass[11pt]{article}
+\usepackage{hdtt2020}
+\usepackage{newpxmath}
+\usepackage{newpxtext}
+
+\title{CSCI 8980 Higher-Dimensional Type Theory\\ Lecture Notes}
+\author{Norihiro Yamada, Nathan Ringo}
+\date{February 11, 2020}
+
+\begin{document}
+
+\maketitle
+
+\section{Extensionality and Intensionality}
+Two commonly used but insufficiently precise terms used to dicuss notions of equality are ``extensional equality'' and ``intensional equality.''
+
+An extensional notion of equality treats two objects that have observationally equivalent properties as being equal. For example, one might consider the sets $A = \left(\{\textrm{single-digit primes}\} \cup \{6\}\right)$ to be extensionally equal to $B = \left(\{\textrm{single-digit numbers}\} - \{x^2 \mid x \in \mathbb{N}\}\right)$, since $\forall n. \left(n \in A \iff n \in B\right)$. Similarly, $x + y$ and $y + x$ would be considered equal, since their results are the same for any $x$ and $y$.
+
+Contrastingly, an intensional notion of equality only considers objects to be equal if their definitions are equal. Often the equality of definitions is checked syntactically; i.e., two definitions are equivalent if and only if they are literally textually identical. With such a definition, neither $A$ and $B$ nor $x + y$ and $y + x$ would \textbf{not} be considered equal,
+
+Theories built on the lambda calculus that claim to use intensional equality will often define this syntactic equality to be after $\alpha$, $\beta$, and $\eta$-normalization. This makes the terms $\lam{x}[\nattype]{x}$ and $\app{\parens{\lam{p}[\funtype{\nattype}{\nattype}]{p}}}{\parens{\lam{q}[\nattype]{q}}}$ equal, and is often useful, since the evaluation rules don't need to be manipulated by the user.
+
+\section{Equalities}
+
+There are several different specific notions of equality used in papers. Some effort is required to determine which one is in use.
+
+\subsection{Judgmental Equality}
+
+A natural idea for implementing equality is to specify pairs of terms to be equated directly by judgments, which convey no information other than the equated terms (and, in the type theories we're studying in this course, their type). The equality judgment is written like:
+
+\[
+	\eqterm{\Gamma}{M}{N}{A}
+\]
+
+Judgmental equality is \emph{silent and substitutional}, in the sense that:
+
+\begin{prooftree*}
+	\hypo{\oftype{\Gamma}{M}{A}}
+	\ellipsis{}{\hspace{0.75in}}
+	\hypo{\eqterm{\Gamma}{M}{N}{A}}
```

Comment:
`\eqtype{\Gamma}{A}{B}` (to replace the type) might make more sense. For this you do not need ellipses, either.